stable_baselines3/common/vec_env/base_vec_env.py

```diff
@@ -162,35 +168,44 @@ def step(self, actions: np.ndarray) -> VecEnvStepReturn:
         self.step_async(actions)
         return self.step_wait()
 
-    def get_images(self) -> Sequence[np.ndarray]:
+    def get_images(self) -> Sequence[Optional[np.ndarray]]:
         """
-        Return RGB images from each environment
+        Return Render output from each environment
         """
         raise NotImplementedError
 
-    def render(self, mode: str = "human") -> Optional[np.ndarray]:
+    def render(self, mode: Optional[str] = None) -> Optional[np.ndarray]:
         """
         Gym environment rendering
 
         :param mode: the rendering type
         """
-        try:
-            imgs = self.get_images()
-        except NotImplementedError:
-            warnings.warn(f"Render not defined for {self}")
-            return
 
-        # Create a big image by tiling images from subprocesses
-        bigimg = tile_images(imgs)
-        if mode == "human":
-            import cv2  # pytype:disable=import-error
+        if mode and self.render_mode != mode:
+            raise ValueError(
+                f"""starting from gym v0.26, render modes are determined during the initialization of the environment.
+                We allow to pass a mode argument to maintain a backwards compatible VecEnv API, but the mode ({mode}) 
+                has to be the same as the environment render mode ({self.render_mode}) whichs is not the case."""
+            )
 
-            cv2.imshow("vecenv", bigimg[:, :, ::-1])
-            cv2.waitKey(1)
-        elif mode == "rgb_array":
+        mode = self.render_mode
```

Comment:
hmm, should be `self.get_attr("render_mode")[0]` I guess.
you can either remove `self.render_mode` or better set it to `self.get_attr("render_mode")[0]` when initializing the envs (and so remove it anyway from the VecEnv constructor).

Currently, having `render_mode`  in VecEnv constructor, is kind of a duplicate, no?